packages/remirror__core/src/extension/extension-decorator.ts

```diff
@@ -154,7 +154,14 @@ export type ExtensionDecoratorOptions<Options extends Shape = EmptyShape> = Defa
 export function extension<Options extends Shape = EmptyShape>(
   options: ExtensionDecoratorOptions<Options>,
 ) {
-  return <Type extends AnyExtensionConstructor>(ReadonlyConstructor: Type): Type => {
+  return <Type extends AnyExtensionConstructor>(
+    ReadonlyConstructor: Type,
+    context: ClassDecoratorContext<Type>,
+  ): Type => {
+    if (context.kind !== 'class') {
+      throw new Error(`The extension decorator can only be used on a class.`);
+    }
```

Comment:
[In the docs](https://remirror.io/docs/concepts/extension#extension-1) we mention that you can use decorators as a normal function

<img width="986" alt="Screenshot 20
23-11-02 at 18 13 35" src="https://github.com/remirror/remirror/assets/2003804/cdd38c94-f834-45fd-8a33-f2ccf9acfcd0">

For this to be true now, we would need to use (at a minimum)

```diff
- extension({ defaultSettings: { color: 'red' } })(ExampleExtension);
+ extension({ defaultSettings: { color: 'red' } })(ExampleExtension, { kind: 'class' });
```

Should we let `context` default to `{ kind: 'class' }` if undefined? Or should we update the docs instead?

Reply:
we let context default to `{ kind: 'class' }`

Reply:
Should an appropriate default be added for all decorators, or just this `extension` one?

Reply:
I think all decorators from Remirror should have default `kind`